src/main/scala/esp/Implementation.scala

```diff
@@ -38,9 +38,21 @@ object ConfigIO {
 
 }
 
+object DmaSize {
+  val SizeByte   = 0.U;
+  val SizeHWord  = 1.U;
+  val SizeWord   = 2.U;
+  val SizeDWord  = 3.U;
+  val Size4Word  = 4.U;
+  val Size8Word  = 5.U;
+  val Size16Word = 6.U;
+  val Size32Word = 7.U;
+}
```

Comment:
A couple of minor improvements to this. Chisel has an [`Enum` object](https://chisel.eecs.berkeley.edu/api/latest/chisel3/util/Enum.html) which is a method for constructing sequences of `UInt`s. It's janky and not type-safe, but it makes what this PR is doing more programmatic. (There's a ["strongly typed enum"](https://github.com/freechipsproject/chisel3/pull/892) which will be added in Chisel 3.2, but that isn't released yet and I don't want to point this repo at snapshot version) I'd also prefer to have the width of the size field not hard-coded to `3`.

How about something like the following:
```scala
import chisel3.util.Enum // Add this up with the other imports
object DmaSize {
  private val enums = Enum(8)
  val Seq(bytes, wordHalf, word, wordDouble, wordQuad, word8, word16, word32) = Enum(8)
  def gen: UInt = enums.head.chiselCloneType
}
```

I change the names here for two reasons: (1) bikeshedding from the repetition of size and (2) you can't use uppercase to construct `val`s in that way (`val Seq(Foo, Bar) = Seq(1, 2)` will error for reasons that I never investigated), and (3) the convention in Scala is that uppercase means "class, object, or trait" while lowercase means "package, member, or function". (2) and (3) matter, but (1) doesn't. Use whatever makes sense to you.